WickedEngine/wiScene.cpp

```diff
@@ -5065,7 +5065,8 @@ namespace wi::scene
 				const AABB& aabb = aabb_objects[objectIndex];
 				if (!ray.intersects(aabb) || (layerMask & aabb.layerMask) == 0)
 					continue;
-
+				if (objectIndex >= objects.GetCount())
+					continue;
```

Comment:
This is not very good, limit the loop count instead, or use break, also not needed to be after the intersect check.